lib/utilization/docker-info.js

```diff
@@ -5,61 +5,162 @@
 
 'use strict'
 
-const logger = require('../logger').child({ component: 'docker-info' })
+const fs = require('node:fs')
+const http = require('node:http')
+const log = require('../logger').child({ component: 'docker-info' })
 const common = require('./common')
 const NAMES = require('../metrics/names')
 const os = require('os')
 let vendorInfo = null
+
 const CGROUPS_V1_PATH = '/proc/self/cgroup'
 const CGROUPS_V2_PATH = '/proc/self/mountinfo'
+const BOOT_ID_PROC_FILE = '/proc/sys/kernel/random/boot_id'
 
 module.exports.getVendorInfo = fetchDockerVendorInfo
 module.exports.clearVendorCache = function clearDockerVendorCache() {
   vendorInfo = null
 }
 
-module.exports.getBootId = function getBootId(agent, callback) {
+module.exports.getBootId = function getBootId(agent, callback, logger = log) {
```

Comment:
it's fine but seems like a lot of signature updates to inject a logger.  if you look through our unit tests we use proxyquire to inject mock loggers

Reply:
I think being able to inject the parameter is a lot cleaner than using proxyquire. We don't need to couple the implementation of the whole module to the tests (the tests needing to know how the logger is imported and used in the code).